conans/test/integration/package_id/package_id_and_confs_test.py

```diff
@@ -16,14 +20,12 @@
     ('["tools.build:cxxflags", "tools.build:cflags"]', PKG_ID_1),
     ('["tools.build:defines"]', PKG_ID_2),
     ('["tools.build:cxxflags", "tools.build:sharedlinkflags"]', PKG_ID_3),
+    ('["user.foo:value"]', PKG_ID_USER_1),
+    ('["user.foo:value", "user.bar:value"]', PKG_ID_USER_2),
+    ('["user.*"]', PKG_ID_USER_3),
```

Comment:
@memsharded this is unrelated to the PR - I initially added this to check that user confs were being handled properly. But note how the generated pkgids are different in these 2 cases, we need to look into that too

Reply:
This is solved by https://github.com/conan-io/conan/pull/15355